src/queue.cpp

```diff
@@ -958,6 +1039,19 @@ cvk_command_kernel::build_batchable_inner(cvk_command_buffer& command_buffer) {
     return CL_SUCCESS;
 }
 
+cl_int cvk_command_kernel::do_post_action() {
+    if (m_kernel->uses_printf()) {
+        auto buffer = m_queue->get_printf_buffer();
+        if (buffer == nullptr) {
+            cvk_error_fn("printf buffer was not created");
```

Comment:
I'm not sure this makes sense as a user-visible error. The only to hit this is if we made a mistake in clvk. Maybe an assert would have been enough/more appropriate.